test/fetch/createFetchMock.ts

```diff
@@ -3,13 +3,17 @@ import {type MockedResponse} from '@Test/fetch/defineMockResponse';
 import {getAutoImplementation} from './getAutoImplementation';
 
 export const createFetchMock = <T>(
-  implementation?: MockedResponse<T>,
+  implementation?: MockedResponse<T> | (() => Promise<MockedResponse<T>>),
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   options?: Record<string, unknown>,
 ): FetchMock => {
   return fetchMock.mockImplementation(async (info, init) => {
     let resolvedImplementation = implementation;
 
+    if (typeof implementation === 'function') {
+      resolvedImplementation = await implementation();
+    }
```

Comment:
Wordt gechecked of de `implementation` param een type promise is?
Zo, ja is het niet beter om `instanceof` te gebruiken?

```suggestion
    if (esolvedImplementation instanceof Promise) {
      resolvedImplementation = await implementation();
    }
```

Reply:
functie die een promise returned kijk maar naar de types